lib/workers/repository/update/branch/index.ts

```diff
@@ -673,7 +674,9 @@ export async function processBranch(
     // skip if we have a non-immediate pr and there is an existing PR,
     // we want to update the PR and skip the Auto merge since status checks aren't done yet
     if (!config.artifactErrors?.length && (!commitSha || config.ignoreTests)) {
-      const mergeStatus = await tryBranchAutomerge(config);
+      const mergeFlag: MergeFlag =
+        config.rebaseWhen === 'conflicted' ? '--ff' : '--ff-only';
```

Comment:
We don't want git awareness (flags) at the worker layer here. Instead the param should be more generic like "allowBehindBase" true/false